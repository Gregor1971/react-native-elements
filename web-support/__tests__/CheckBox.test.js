import React from 'react';
import {
  Text,
  TouchableHighlight,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import parseSamples, {
  assignRef,
  genRefId,
} from 'enzyme-styleguidist-sample-parser';
import options, {
  snapShot,
  buildJsxForGuideMethod,
  ensureCalled,
  onlyEnsureCalled,
  onlySnapshots,
} from '../samples';
import { CheckBox as Component } from '../../src';

const props = {
  noProps: {
    component: Component,
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'component, title, checked, & onIconPress': {
    component: Component,
    props: {
      component: TouchableWithoutFeedback,
      checked: true,
      title: 'Title',
      onIconPress: () => {
        let x = 7;
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      getJsxString: () => {
        return `const TouchableWithoutFeedback = RN.TouchableWithoutFeedback;
  <CheckBox 
    component={TouchableWithoutFeedback}
    title="title"
    checked={state.checked}
    onIconPress={() => {
      setState({ checked: !state.checked });
    }}
  />`;
      },
    },
  },
  'iconRight, title, checked, & onIconPress': {
    component: Component,
    props: {
      iconRight: true,
      title: 'Title',
      checked: true,
      onIconPress: () => {
        let x = 7;
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `initialState = { checked: false };`,
      getJsxString: () => {
        return `<CheckBox
    iconRight
    title="title"
    checked={state.checked}
    onIconPress={() => {
      setState({ checked: !state.checked });
    }}
  />`;
      },
    },
  },
  'right, title, checked, & onIconPress': {
    component: Component,
    props: {
      right: true,
      title: 'Title',
      checked: true,
      onIconPress: () => {
        let x = 7;
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `initialState = { checked: false };`,
      getJsxString: () => {
        return `<CheckBox
  right
  title="title"
  checked={state.checked}
  onIconPress={() => {
    setState({ checked: !state.checked });
  }}
/>`;
      },
    },
  },
  'center, title, checked, & onIconPress': {
    component: Component,
    props: {
      center: true,
      title: 'Title',
      checked: true,
      onIconPress: () => {
        let x = 7;
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `initialState = { checked: false };`,
      getJsxString: () => {
        return `<CheckBox
    center
    title="title"
    checked={state.checked}
    onIconPress={() => {
      setState({ checked: !state.checked });
    }}
  />`;
      },
    },
  },
  containerStyle: {
    component: Component,
    props: { containerStyle: { backgroundColor: '#071' } },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'textStyle & title': {
    component: Component,
    props: { textStyle: { backgroundColor: '#071' }, title: 'Title' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'onLongPress, title, checked, & onIconPress': {
    component: Component,
    props: {
      title: 'Title',
      checked: true,
      onIconPress: () => {
        let x = 7;
      },
      onLongPress: () => {
        let y = 8;
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `initialState = { checked: false };`,
      getJsxString: () => {
        return `<CheckBox
    title="title"
    checked={state.checked}
    onIconPress={() => {
      setState({ checked: !state.checked });
    }}
    onLongPress={() => {
      console.log("pressed, long");
    }}
  />`;
      },
    },
  },
  'onLongIconPress, title, checked, & onIconPress': {
    component: Component,
    props: {
      title: 'Title',
      checked: true,
      onIconPress: () => {
        let x = 7;
      },
      onLongPress: () => {
        let y = 8;
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `initialState = { checked: false };`,
      getJsxString: () => {
        return `<CheckBox
    title="title"
    checked={state.checked}
    onIconPress={() => {
      setState({ checked: !state.checked });
    }}
    onLongIconPress={() => {
      console.log("pressed, long");
    }}
  />`;
      },
    },
  },
  'onPress, title, checked, & onIconPress': {
    component: Component,
    props: {
      title: 'Title',
      checked: true,
      onIconPress: () => {
        let x = 7;
      },
      onPress: () => {
        let y = 8;
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `initialState = { checked: false };`,
      getJsxString: () => {
        return `<CheckBox
    title="title"
    checked={state.checked}
    onIconPress={() => {
      setState({ checked: !state.checked });
    }}
    onPress={() => {
      setState({ checked: !state.checked });
    }}
  />`;
      },
    },
  },
  checkedIcon: {
    component: Component,
    props: { checked: true, checkedIcon: 'bell-o' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  uncheckedIcon: {
    component: Component,
    props: { uncheckedIcon: 'bell-slash-o' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  checkedColor: {
    component: Component,
    props: { checked: true, checkedColor: '#22d' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  uncheckedColor: {
    component: Component,
    props: { uncheckedColor: '#d22' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'checkedTitle, title, checked, & onIconPress': {
    component: Component,
    props: {
      checkedTitle: 'title - is checked',
      title: 'Title',
      checked: true,
      onIconPress: () => {
        let x = 7;
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      script: `initialState = { checked: false };`,
      getJsxString: () => {
        return `<CheckBox
    checkedTitle="title - is checked"
    title="title"
    checked={state.checked}
    onIconPress={() => {
      setState({ checked: !state.checked });
    }}
  />`;
      },
    },
  },
  fontFamily: {
    component: Component,
    props: { title: 'Title', fontFamily: 'Courier New' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
};

const samples = {
  props,
};

parseSamples(
  {
    Input: { sectionComponents: { CheckBox: { samples } } },
  },
  options
);
