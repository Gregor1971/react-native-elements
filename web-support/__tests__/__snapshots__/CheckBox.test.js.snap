// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input: CheckBox: [ prop: center, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <[object Object]
      color="green"
      name="check-square-o"
      onPress={[Function]}
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    >
      Title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: [ prop: checkedColor ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="#22d"
      name="check-square-o"
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: [ prop: checkedIcon ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="green"
      name="bell-o"
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: [ prop: checkedTitle, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="green"
      name="check-square-o"
      onPress={[Function]}
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    >
      title - is checked
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: [ prop: component, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableWithoutFeedback
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="green"
      name="check-square-o"
      onPress={[Function]}
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    >
      Title
    </TextElement>
  </View>
</TouchableWithoutFeedback>
`;

exports[`Input: CheckBox: [ prop: containerStyle ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      Object {
        "backgroundColor": "#071",
      },
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="#bfbfbf"
      name="square-o"
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: [ prop: fontFamily ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="#bfbfbf"
      name="square-o"
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          Object {
            "fontFamily": "Courier New",
          },
        ]
      }
    >
      Title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: [ prop: iconRight, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    >
      Title
    </TextElement>
    <[object Object]
      color="green"
      name="check-square-o"
      onPress={[Function]}
      size={24}
    />
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: [ prop: noProps ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="#bfbfbf"
      name="square-o"
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: [ prop: onLongIconPress, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onLongPress={[Function]}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="green"
      name="check-square-o"
      onPress={[Function]}
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    >
      Title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: [ prop: onLongPress, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onLongPress={[Function]}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="green"
      name="check-square-o"
      onPress={[Function]}
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    >
      Title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: [ prop: onPress, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onPress={[Function]}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="green"
      name="check-square-o"
      onPress={[Function]}
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    >
      Title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: [ prop: right, title, checked, & onIconPress ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        Object {
          "justifyContent": "flex-end",
        },
        false,
      ]
    }
  >
    <[object Object]
      color="green"
      name="check-square-o"
      onPress={[Function]}
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    >
      Title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: [ prop: textStyle & title ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="#bfbfbf"
      name="square-o"
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          Object {
            "backgroundColor": "#071",
          },
          undefined,
        ]
      }
    >
      Title
    </TextElement>
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: [ prop: uncheckedColor ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="#d22"
      name="square-o"
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`Input: CheckBox: [ prop: uncheckedIcon ] enzyme-shallow: snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  style={
    Array [
      103,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        102,
        false,
        false,
      ]
    }
  >
    <[object Object]
      color="#bfbfbf"
      name="bell-slash-o"
      size={24}
    />
    <TextElement
      style={
        Array [
          104,
          undefined,
          undefined,
        ]
      }
    />
  </View>
</TouchableOpacity>
`;
